File: webservice/Frikollection_Api/Frikollection_Api/DTOs/Collection/CollectionPreviewDto.cs
using Frikollection_Api.DTOs.Product;
using Frikollection_Api.DTOs.User;

namespace Frikollection_Api.DTOs.Collection
{
    public class CollectionPreviewDto
    {
        public string Name { get; set; }
        public bool? Private { get; set; }
        public DateOnly? CreationDate { get; set; }
        public UserPreviewDto? User { get; set; }
        public ICollection<ProductDto>? Products { get; set; } = new List<ProductDto>();
    }
}

File: mobile_desktop_app/FrikollectionMobileDesktop/composeApp/src/commonMain/kotlin/org/example/frikollection_mobile_desktop/models/collection/CollectionPreviewDto.kt
package org.example.frikollection_mobile_desktop.models.collection

import kotlinx.serialization.Serializable
import org.example.frikollection_mobile_desktop.models.product.ProductDto

@Serializable
data class CollectionPreviewDto(
    val name: String,
    val private: Boolean? = null,
    val creationDate: String? = null,
    val user: UserPreviewDto? = null,
    val products: List<ProductDto> = emptyList()
)

@Serializable
data class UserPreviewDto(
    val username: String
)